File: applications/luci-olsr/luasrc/model/cbi/olsr/olsrdhna.lua
--[[
LuCI - Lua Configuration Interface

Copyright 2008 Steven Barth <steven@midlink.org>

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

	http://www.apache.org/licenses/LICENSE-2.0

$Id$
]]--

mh = Map("olsrd", translate("OLSR - HNA-Announcements"))


hna4 = mh:section(TypedSection, "Hna4", translate("Hna4"), translate("Both values must use the dotted decimal notation."))
hna4.addremove = true
hna4.anonymous = true
hna4.template  = "cbi/tblsection"

net4 = hna4:option(Value, "netaddr", translate("Network address"))
net4.datatype = "ip4addr"
net4.placeholder = "15.15.0.0"
msk4 = hna4:option(Value, "netmask", translate("Netmask"))
msk4.datatype = "ip4addr"
msk4.placeholder = "255.255.255.0"

hna6 = mh:section(TypedSection, "Hna6", translate("Hna6"), translate("IPv6 network must be given in full notation, " ..
	"prefix must be in CIDR notation."))
hna6.addremove = true
hna6.anonymous = true
hna6.template  = "cbi/tblsection"

net6 = hna6:option(Value, "netaddr", translate("Network address"))
net6.datatype = "ip6addr"
net6.placeholder = "fec0:2200:106:0:0:0:0:0"
msk6 = hna6:option(Value, "prefix", translate("Prefix"))
msk6.datatype = "integer"
msk6.placeholder = "48"


return mh

File: applications/luci-app-olsr/htdocs/luci-static/resources/view/olsr/frontend/olsrdhna.js
'use strict';
'require view';
'require form';
'require uci';
'require ui';

return view.extend({
	load() {
		return Promise.all([uci.load('olsrd')]);
	},
	render() {
		let ipv = uci.get_first('olsrd', 'olsrd', 'IpVersion') || '4';

		let mh = new form.Map('olsrd', _('OLSR - HNA-Announcements'), _('Hosts in an OLSR routed network can announce connectivity ' + 'to external networks using HNA messages.'));

		if (ipv === '6and4' || ipv === '4') {
			let hna4 = mh.section(form.TypedSection, 'Hna4', _('Hna4'), _('Both values must use the dotted decimal notation.'));
			hna4.addremove = true;
			hna4.anonymous = true;
			hna4.template = 'cbi/tblsection';

			let net4 = hna4.option(form.Value, 'netaddr', _('Network address'));
			net4.datatype = 'ip4addr';
			net4.placeholder = '10.11.12.13';
			net4.default = '10.11.12.13';
			let msk4 = hna4.option(form.Value, 'netmask', _('Netmask'));
			msk4.datatype = 'ip4addr';
			msk4.placeholder = '255.255.255.255';
			msk4.default = '255.255.255.255';
		}

		if (ipv === '6and4' || ipv === '6') {
			let hna6 = mh.section(form.TypedSection, 'Hna6', _('Hna6'), _('IPv6 network must be given in full notation, ' + 'prefix must be in CIDR notation.'));
			hna6.addremove = true;
			hna6.anonymous = true;
			hna6.template = 'cbi/tblsection';

			let net6 = hna6.option(form.Value, 'netaddr', _('Network address'));
			net6.datatype = 'ip6addr';
			net6.placeholder = 'fec0:2200:106:0:0:0:0:0';
			net6.default = 'fec0:2200:106:0:0:0:0:0';
			let msk6 = hna6.option(form.Value, 'prefix', _('Prefix'));
			msk6.datatype = 'range(0,128)';
			msk6.placeholder = '128';
			msk6.default = '128';
		}
		return mh.render();
	},
});
